jerry-core/ecma/base/ecma-globals.h

```diff
@@ -115,6 +120,26 @@ typedef uint32_t ecma_value_t;
 #define ECMA_VALUE_SHIFT 3
 
 /**
+ * Flag for simple values
+ */
+#define ECMA_SIMPLE_VALUE_FLAG 0x8u
```

Comment:
In this case I would prefer the full value because of the name, i.e.:

    #define ECMA_DIRECT_INTEGER_VALUE (0u << ECMA_VALUE_SHIFT) | ECMA_TYPE_DIRECT